spk/sickchill/src/service-setup.sh

```diff
@@ -1,53 +1,61 @@
+VIRTUALENV="${PYTHON_BIN}/virtualenv"
 
+VENV="${SYNOPKG_PKGHOME}/.venv"
+PIP="${VENV}/bin/pip3"
+PYTHON="${VENV}/bin/python3"
+
+PATH="${VENV}/bin:${PATH}:${SYNOPKG_PKGDEST}/bin:${PYTHON_BIN}:${PATH}"
+
+SICKCHILL="${VENV}/bin/SickChill.py"
 
 GROUP="sc-download"
 
-SERVICE_COMMAND="${PYTHON} ${SC_BINARY} --daemon --nolaunch --pidfile ${PID_FILE} --config ${SC_CFG_FILE} --datadir ${SC_DATA_DIR}"
+PIDFILE="${SYNOPKG_PKGTMP}/${SYNOPKG_PKGNAME}.pid"
+
+SERVICE_COMMAND="LANG=en_US.UTF-8 ${PYTHON} ${SICKCHILL} --daemon --nolaunch --pidfile=${PIDFILE} --datadir=${SYNOPKG_PKGVAR} --port=${SYNOPKG_PKGPORT}"
+
+SVC_CWD="${SYNOPKG_PKGDEST}/"
+HOME="${SYNOPKG_PKGDEST}/"
+
 
 set_config() {
-    . ${SYNOPKG_PKGDEST}/share/git_data
+    CFG_FILE="${SYNOPKG_PKGVAR}/config.ini"
 
-    if [ -f "${SC_CFG_FILE}" ]; then
+    if [ -f "${CFG_FILE}" ]; then
         if [ -n "${wizard_username}" ] && [ -n "${wizard_password}" ]; then
-            sed -i "/^\s*web_username\s*=/s/\s*=\s*.*/ = ${wizard_username}/" ${SC_CFG_FILE}
-            sed -i "/^\s*web_password\s*=/s/\s*=\s*.*/ = ${wizard_password}/" ${SC_CFG_FILE}
+            sed -i "/^\s*web_username\s*=/s/\s*=\s*.*/ = ${wizard_username}/" ${CFG_FILE}
+            sed -i "/^\s*web_password\s*=/s/\s*=\s*.*/ = ${wizard_password}/" ${CFG_FILE}
         fi
-        sed -i "/^\s*branch\s*=/s/\s*=\s*.*/ = ${SC_CONFIG_GIT_BRANCH}/" ${SC_CFG_FILE}
-        sed -i "/^\s*cur_commit_hash\s*=/s/\s*=\s*.*/ = ${SC_CONFIG_GIT_COMMIT_HASH}/" ${SC_CFG_FILE}
-        sed -i "/^\s*cur_commit_branch\s*=/s/\s*=\s*.*/ = ${SC_CONFIG_GIT_COMMIT_BRANCH}/" ${SC_CFG_FILE}
     else
-        mkdir -p ${SC_DATA_DIR}
-        cat << EOF > ${SC_CFG_FILE}
+        mkdir -p ${SYNOPKG_PKGVAR}
+        cat << EOF > ${CFG_FILE}
 [General]
 web_username = ${wizard_username}
 web_password = ${wizard_password}
-branch = ${SC_GIT_BRANCH}
-cur_commit_hash = ${SC_GIT_COMMIT_HASH}
-cur_commit_branch = ${SC_GIT_COMMIT_BRANCH}
 EOF
     fi
 }
 
 service_postinst() {
     # Create a Python virtualenv
-    ${VIRTUALENV} --system-site-packages ${SYNOPKG_PKGDEST}/env >>${INST_LOG}
+    ${VIRTUALENV} --system-site-packages ${VENV}
+
+    # If sickchill is not installed via a wheel, site-packages cannot be included due to shadowing.
+    # ${VIRTUALENV}  ${VENV}
 
     # Install the wheels
-    ${PIP} install --no-deps --no-index -U --force-reinstall -f ${SYNOPKG_PKGDEST}/share/wheelhouse ${SYNOPKG_PKGDEST}/share/wheelhouse/*.whl >> ${INST_LOG} 2>&1
+    ${PIP} install --no-index -U --force-reinstall -f ${SYNOPKG_PKGDEST}/share/wheelhouse ${SYNOPKG_PKGDEST}/share/wheelhouse/*.whl
+
+    # If package inscludes all deps, need --no-deps here
+     ${PIP} install --no-deps --no-index -U --force-reinstall -f ${SYNOPKG_PKGDEST}/share/wheelhouse ${SYNOPKG_PKGDEST}/share/wheelhouse/*.whl
 
     if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
         set_config
     fi
+
+    set_unix_permissions "${SYNOPKG_PKGHOME}"
+    set_unix_permissions "${SYNOPKG_PKGVAR}"
```

Comment:
If needed, this is for DSM<7 only.

Reply:
Yes, the venv is in SYNO_PKGHOME and needs to be writable by sc-sickchill. On dsm7 this will just be ignored, but I'd like to make the package work for both dsm6 and 7.

Is syno_pkghome defined for dsm6 in the new toolchain?